src/main/java/com/google/api/codegen/util/ProtoParser.java

```diff
@@ -129,62 +134,72 @@ public OperationData getLongRunningOperation(Method method) {
         (resourceSet, baseNameToSet) -> resourceSet.toBuilder().setName(baseNameToSet).build());
   }
 
-  /* Return a Map of names to their corresponding Resource or ResourceSet element. */
-  private <T> List<T> getResourceOrSetDefs(
-      ProtoFile protoFile,
+  /* Return a Map of Resource or ResourceSet elements to their containing ProtoFile. */
+  private <T> Map<T, ProtoFile> getResourceOrSetDefs(
+      List<ProtoFile> protoFiles,
       DiagCollector diagCollector,
       GeneratedExtension<FileOptions, List<T>> fileExtension,
       GeneratedExtension<FieldOptions, T> fieldExtension,
       Function<T, String> getNameFunc,
       BiFunction<T, String, T> setNameFunc) {
-    Map<String, T> resourceSetDefs = new LinkedHashMap<>();
-    List<T> resourcesAtFileLevel = getProtoExtension(protoFile, fileExtension);
-    // Get definitions from protofile options.
-    if (resourcesAtFileLevel != null) {
-      for (T definition : resourcesAtFileLevel) {
-        String baseName = getNameFunc.apply(definition);
-        if (Strings.isNullOrEmpty(baseName)) {
-          diagCollector.addDiag(
-              Diag.error(
-                  SimpleLocation.TOPLEVEL,
-                  "There is a %s option with"
-                      + " no name defined in proto file %s. %s.name is required.",
-                  fileExtension.getDescriptor().getFullName(),
-                  protoFile.getFullName(),
-                  fileExtension.getDescriptor().getFullName()));
-          continue;
-        }
+    ImmutableMap.Builder<T, ProtoFile> definitions = ImmutableMap.builder();
+
+    for (ProtoFile protoFile : protoFiles) {
 
-        if (resourceSetDefs.containsKey(baseName)) {
-          diagCollector.addDiag(
-              Diag.error(
-                  SimpleLocation.TOPLEVEL,
-                  "Multiple %s defintions with the name"
-                      + " %s are defined in proto file %s. Values for %s.name must be unique.",
-                  fieldExtension.getDescriptor().getFullName(),
-                  baseName,
-                  protoFile.getFullName(),
-                  fieldExtension.getDescriptor().getFullName()));
-          continue;
+      // Maps base names to Resource[Sets].
+      Map<String, T> localDefs = new LinkedHashMap<>();
+      // Get definitions from protofile options.
+      List<T> resourcesAtFileLevel = getProtoExtension(protoFile, fileExtension);
+      if (resourcesAtFileLevel != null) {
```

Comment:
Here and below, you can `if (resourcesAtFileLevel == null) continue;` to reduce nesting.

Reply:
Hmm, but I don't want to continue if `(resourcesAtFileLevel == null)`? I still want to parse the protofile for more Resources and ResourceSets down below.

Reply:
Derp you're right. I got confused by the nesting :P